designcompose/src/main/java/com/android/designcompose/squoosh/SquooshTreeBuilder.kt

```diff
@@ -192,22 +182,24 @@ internal fun resolveVariantsRecursively(
 
         // Do we have an override style? This is style data which we should apply to the final style
         // even if we're swapping out our view definition for a variant.
-        if (componentInfo.overrides.isPresent) {
-            val overrides = componentInfo.overrides.get()
-            if (overrides.style.isPresent) {
-                overrideStyle = overrides.style.get()
+        if (v.componentInfo.hasOverrides()) {
+            val overrides = v.componentInfo.overrides
+            if (overrides.hasStyle()) {
+                overrideStyle = overrides.style
             }
         }
 
         // See if we have a variant replacement; this only happens for component instances (for both
         // interaction-driven and customization-driven variant changes).
 
         // First ask the interaction state if there's a variant we should render instead.
-        view = interactionState.squooshNodeVariant(v.id, customizations.getKey(), document) ?: v
+        interactionState.squooshNodeVariant(v.id, customizations.getKey(), document)?.let {
+            view = it
+        }
 
         // If we didn't replace the component because of an interaction, we might want to replace it
         // because of a variant customization.
-        if (view.name == v.name) {
+        if (view.name == v.name) { // TODO: why we don't check if view == v????
```

Comment:
Not sure if this means `view == v` and below if I can assume `view.componentInfo` is actually the same with `v.componentInfo`

Reply:
This code really isn't clear, it's hard to tell what `v` and `view` are. @rylin8, could you take a look at this and maybe clean it up with some better variable nmames?

Reply:
Done